Doc/whatsnew/3.12.rst

```diff
@@ -98,6 +98,57 @@ Deprecated
 ==========
 
 
+Pending Removal in Python 3.13
+==============================
+
+The following APIs have been deprecated in earlier Python releases,
+and will be removed in Python 3.13.
+
+* :mod:`aifc` (:pep:`594`)
+* :mod:`audioop` (:pep:`594`)
+* :mod:`cgi` (:pep:`594`)
+* :mod:`cgitb` (:pep:`594`)
+* :mod:`chunk` (:pep:`594`)
+* :class:`configparser.LegacyInterpolation` (:issue:`46607`)
+* :mod:`crypt` (:pep:`594`)
+* :mod:`imghdr` (:pep:`594`)
+* :func:`locale.getdefaultlocale` (:issue:`46659`)
+* :mod:`mailcap` (:pep:`594`)
+* :mod:`msilib` (:pep:`594`)
+* :mod:`nis` (:pep:`594`)
+* :mod:`nntplib` (:pep:`594`)
+* :mod:`ossaudiodev` (:pep:`594`)
+* :mod:`pipes` (:pep:`594`)
+* :mod:`sndhdr` (:pep:`594`)
+* :mod:`spwd` (:pep:`594`)
+* :mod:`sunau` (:pep:`594`)
+* :mod:`telnetlib` (:pep:`594`)
+* :meth:`turtle.RawTurtle.settiltangle` (:issue:`45837`)
+* :func:`unittest.findTestCases` (:issue:`5846`)
+* :func:`unittest.makeSuite` (:issue:`5846`)
+* :func:`unittest.getTestCaseNames` (:issue:`5846`)
+* :mod:`uu` (:pep:`594`)
+* :class:`webbrowser.MacOSX` (:issue:`42255`)
+* :mod:`xdrlib` (:pep:`594`)
+
+Pending Removal in Future Versions
+==================================
+
+The following APIs were deprecated in earlier Python versions and will be removed,
+although there is currently no date scheduled for their removal.
+
+* :class:`typing.Text` (:gh:`92332`)
+
+* Currently Python accepts numeric literals immediately followed by keywords,
+  for example ``0in x``, ``1or x``, ``0if 1else 2``.  It allows confusing
+  and ambiguous expressions like ``[0x1for x in y]`` (which can be
+  interpreted as ``[0x1 for x in y]`` or ``[0x1f or x in y]``).
+  A deprecation warning is raised if the numeric literal is
```

Comment:
In Python 3.12, I get a SyntaxWarning on ``0in x``.

```
>>> x=[]
>>> 0in x
<stdin>:1: SyntaxWarning: invalid decimal literal
False
```

Reply:
Aha, the warning was changed just in time for the 3.11 beta :)

https://github.com/python/cpython/pull/91980

I've updated the text.